### DISCUSSION TITLE
lag and many player

### DISCUSSION THREAD

**Discussion by xengine-qyt** [azerothcore/azerothcore-wotlk#18977] [General]
Is there a recommendation for the number of players and server hardware configuration for AzerothCore? I noticed that with 8 CPUs and 16GB of memory, there is significant lag when there are more than 6000 players. However, the CPU and memory usage don't seem to be high. I don't know why this is happening.

![image](https://github.com/azerothcore/azerothcore-wotlk/assets/37919215/10bba529-a733-4f5a-8139-ce30aeb8f672)

my configure :
ThreadPool = 8
UseProcessors = 0
ProcessPriority = 0
MapUpdate.Threads = 8

**Comment by sudlud** [comment]
did you set your MapThreads to at least 8 in the config? also what is the output of
`.server debug`
while running with 6000 players

**Reply by xengine-qyt** [reply]
> MapThreads

It's already set.,MapUpdate.Threads = 8 

AC> .server debug
AC> AzerothCore rev. 12ea82a57f53 2024-05-26 23:41:04 +0800 (Playerbot branch) (Unix, RelWithDebInfo, Static)
Connected players: 0. Characters in world: 7000.
Connection peak: 0.
Server uptime: 6 hour(s) 19 minute(s) 33 second(s)
Update time diff: 1704ms. Last 500 diffs summary:
- Mean: 1641ms
- Median: 1639ms
- Percentiles (95, 99, max): 1852ms, 1947ms, 2050ms
Using SSL version: OpenSSL 3.0.2 15 Mar 2022 (library: OpenSSL 3.0.2 15 Mar 2022)
Using Boost version: 1.74.0
Using CMake version: 3.22.1
Using MySQL version: 80036
Found MySQL Executable: /usr/bin/mysql
Compiled on: Linux 6.5.0-35-generic
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Enabled
maps directory located in /home/azero/azeroth-server/data/maps. Total size: 291014951 bytes
vmaps directory located in /home/azero/azeroth-server/data/vmaps. Total size: 658130721 bytes
mmaps directory located in /home/azero/azeroth-server/data/mmaps. Total size: 2192910844 bytes
Default DBC locale: enUS.
All available DBC locales: enUS 
Using World DB: ACDB 335.11-dev
Using Playerbots DB Revision: 
Latest LoginDatabase update: 2024_01_20_00.sql
Latest CharacterDatabase update: trasmorg.sql
Latest WorldDatabase update: world_playerbots_rpg_races.sql
LoginDatabase queue size: 0
CharacterDatabase queue size: 0
WorldDatabase queue size: 0
> List enable modules:
PlayerbotsDatabase queue size: 0
> List enable modules:
- mod-arena-3v3-solo-queue
- mod-boss-announcer
- mod-breaking-news-override
- mod-eluna
- mod-guildhouse
- mod-learn-spells
- mod-money-for-kills
- mod-npc-all-mounts
- mod-npc-gambler
- mod-playerbots
- mod-top-arena
- mod-transmog
- mod-weekend-xp
- mod-world-chat

**Reply by sudlud** [reply]
Should most likely be related to the playerbots.

Did you try which count runs stable? Like 1000, 2000 etc

**Reply by sudlud** [reply]
Also are all your bots in one map?

**Reply by xengine-qyt** [reply]
I have tested it, and it runs smoothly with fewer than 5000 players. However, once it exceeds 6000, it becomes unplayable with noticeable lag, and upgrading and adjusting the configuration doesn't help. I don't know why. The players are not on the same map. These bots are divided into Alliance and Horde, and they are spread across maps ranging from level 1 to 35.Fewer than 100 people per map.

**Reply by xengine-qyt** [reply]
Can each server only support a maximum of around 5000 people?

**Reply by sudlud** [reply]
> I have tested it, and it runs smoothly with fewer than 5000 players. However, once it exceeds 6000, it becomes unplayable with noticeable lag, and upgrading and adjusting the configuration doesn't help. I don't know why. The players are not on the same map. These bots are divided into Alliance and Horde, and they are spread across maps ranging from level 1 to 35.Fewer than 100 people per map.
> 
> 

Well then that's your current limit when you want to use playerbots and your current hardware setup.

The server can support more than 5000 players, but this is only true vor clean AC and real players, using playerbots is always a different topic.

Using a CPU with more single core power could help maybe.

Also a map is e.g. kalimdor or eastern kingdoms, e.g. all players in kalimdor will be updated in a single thread all after another, so this is what might lead to the lags on higher player numbers

**Reply by xengine-qyt** [reply]
Well,.I don't know either
